memcache_proxy/main/memcached2g.go

```diff
@@ -48,42 +47,29 @@ func init() {
 func main() {
 	flag.Parse()
 
-	// Apply the overrides.
-	// The final order of overrides will be:
-	// 1. those cmd line parameters
-	// 2. the matching standard GAE env variables (see below)
-	// 3. the VM metadata entries ("instance/attributes/gae_project" for example.)
-	if application != nil {
-		if err := os.Setenv("GAE_LONG_APP_ID", *application); err != nil {
-			panic(err)
-		}
-	}
-	if module != nil {
-		if err := os.Setenv("GAE_MODULE_NAME", *module); err != nil {
-			panic(err)
-		}
-	}
-	if version != nil {
-		if err := os.Setenv("GAE_MODULE_VERSION", *version); err != nil {
-			panic(err)
-		}
-	}
-	if instance != nil {
-		if err := os.Setenv("GAE_MODULE_INSTANCE", *instance); err != nil {
-			panic(err)
-		}
+	// Apply environment variable overrides, in order of preference:
+	// 1. Command-line flags.
+	// 2. Matching standard GAE env variables (see below)
+	// 3. VM metadata entries (e.g., "instance/attributes/gae_project")
+	overrideEnv("GAE_LONG_APP_ID", *application)
+	overrideEnv("GAE_MODULE_NAME", *module)
+	overrideEnv("GAE_MODULE_VERSION", *version)
+	overrideEnv("GAE_MODULE_INSTANCE", *instance)
+	overrideEnv("API_HOST", *apiHost)
+	overrideEnv("API_PORT", *apiPort)
+
+	ctx := gaeint.BackgroundContext()
+	if err := dtog.StartSync(ctx, *bindingAddr); err != nil {
+		log.Fatalf("StartSync: %v", err)
 	}
-	if apiHost != nil {
-		if err := os.Setenv("API_HOST", *apiHost); err != nil {
-			panic(err)
-		}
+}
+
+// overrideEnv sets an environment variable if the desired value is not empty.
+func overrideEnv(k, v string) {
```

Comment:
I would probably make the flags all default to the value of os.Getenv, then always re-set, as it simplifies the code and one can avoid the "override" concept as the relationship between env vars and flags will become more clear.